### PR TITLE
Support show create db

### PR DESCRIPTION
**Pull request by shenli**
We could run mydumper on TiDB now.
Fix: https://github.com/pingcap/tidb/issues/1851
See: http://dev.mysql.com/doc/refman/5.7/en/show-create-database.html

### CODE REVIEW

File: executor/show.go [hunk 1]
```diff
@@ -488,6 +490,25 @@ func (e *ShowExec) fetchShowCreateTable() error {
...
+
+	db, ok := e.is.SchemaByName(e.DBName)
+	if !ok {
+		return infoschema.ErrDatabaseNotExists.Gen("Unknown database '%s'", e.DBName.O)
+	}
+
+	var buf bytes.Buffer
+	buf.WriteString(fmt.Sprintf("CREATE DATABASE `%s`", db.Name.O))
+	if s := db.Charset; len(s) > 0 {
+		buf.WriteString(fmt.Sprintf(" DEFAULT CHARSET=%s", s))
```
Comment (zimulala) [inline executor/show.go]: It prints `/*!40100 DEFAULT CHARACTER SET utf8 */` in the MySQL. Do we need
 to be the same as MySQL?

**Comment by tiancaiamao** [comment]
rest LGTM

File: executor/show.go (hunk 1 above)
Comment (shenli) [inline executor/show.go]: No we just want to run mydumper. I have tested it.

Comment (ngaut) [inline executor/show.go]: I prefer we print the same output.

**Comment by zimulala** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**